File: PYTHON/garage.py
class Garage:
    def __init__(self, nama_garasi="", luas_garasi=""):
        self._nama_garasi = nama_garasi
        self._luas_garasi = luas_garasi
        self._daftar_kendaraan = []

    @property
    def nama_garasi(self):
        return self._nama_garasi

    @property
    def luas_garasi(self):
        return self._luas_garasi

    def tambah_kendaraan(self, kendaraan):
        self._daftar_kendaraan.append(kendaraan)

    def tampilkan_info_kendaraan(self):
        print(f"Info Kendaraan di garasi {self.nama_garasi}:")
        for kendaraan in self._daftar_kendaraan:
            print(kendaraan.display_info())

File: CPP/garage.cpp
#pragma once
#include <string>
#include <vector>
#include <iostream>
#include "vehicle.cpp" 

class Garage {
private:
    std::string _nama_garasi;
    int _luas_garasi;
    std::vector<Vehicle*> _daftar_kendaraan;

public:
    Garage(std::string nama_garasi = "", int luas_garasi = 0);
    
    std::string getNamaGarasi();
    int getLuasGarasi();
    
    void tambahKendaraan(Vehicle* kendaraan); 
    void tampilkanInfoKendaraan();
};

Garage::Garage(std::string nama_garasi, int luas_garasi)
    : _nama_garasi(nama_garasi), _luas_garasi(luas_garasi) {}

std::string Garage::getNamaGarasi() {
    return _nama_garasi;
}

int Garage::getLuasGarasi() {
    return _luas_garasi;
}

void Garage::tambahKendaraan(Vehicle* kendaraan) { 
    _daftar_kendaraan.push_back(kendaraan);
}

void Garage::tampilkanInfoKendaraan() {
    std::cout << "Info Kendaraan di garasi " << _nama_garasi << ":\n";
    for (Vehicle* kendaraan : _daftar_kendaraan) {
        std::cout << kendaraan->displayInfo() << std::endl;
    }
}
